### PR TITLE
Restructure and refine documentation + add missing tests

### PR DESCRIPTION
**Pull request by gudmdharalds**
## Description

The aim of this pull request is to refine and restructure the documentation for the project.

## Pre-deploy checklist

-  [x] Refine files:
   - [x] README.md
   - [x] docs/SETUP.md
     - [x] Add environmental variables
     - [x] Add configuration files
   - [x] docs/ARCHITECTURE.md
   - [x] docs/CONTRIBUTING.md
   - [x] docs/DEBUGGING.md
   - [x] docs/TESTING.md
   - [x] docs/RELEASING.md
   - [x] docs/SECURITY.md
   - [x] docs/CHANGELOG.md
   - [x] Update .github/PULL_REQUEST_TEMPLATE.md
   - [x] Ensure links are correct.
- [x] Add missing tests:
  - [x] Dependaban check
  - [x] Format:check
- [x] Merge #1577 before this one.
  - [x] Remove `CHANGELOG.md` from `.prettierignore`
- [x] Automated tests pass.

## Deploy Notes

None.

## Steps to Test

None.

### CODE REVIEW

File: docs/SETUP.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+### Installation for end-users
+
+First, install the package:
+
+```bash
+npm install -g @automattic/vip
+```
+
+The above command may fail in Linux if you do not have `make` and `g++` installed. In that case, install them first:
```
Comment (sjinks) [inline docs/SETUP.md]: This is no longer necessary.

File: docs/SETUP.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+TODO:
+
+- `A`: Used to ...
+- `B`: Used to ...
+
+## Updating dependencies
+
+### NPM packages
+
+Dependencies generally should be updated by merging [pull requests from dependabot](https://github.com/Automattic/vip-cli/pulls/app%2Fdependabot). However, great care should be taken before merging as updating dependencies can cause fatal errors with the `vip-cli` application, or make it unstable or misbehave. There can also be effects to the deployment process. This can happen silently.
```
Comment (sjinks) [inline docs/SETUP.md]: We don't have Dependabot configured for this repo :-( I update everything manually :-(

File: docs/SETUP.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+### Upgrading Node Version
+
+To upgrade Node, follow these steps:
+
+1. Update [.nvmrc](https://github.com/Automattic/vip-cli/blob/trunk/.nvmrc) to the new version and open a pull request.
+1. Address any test failures.
+1. Update the Node.js version locally and test before merging into the repository.
```
Comment (sjinks) [inline docs/SETUP.md]: 1. Update `engines.node` in `package.json`.

File: docs/SETUP.md (hunk 2 above)
Comment (gudmdharalds) [reply docs/SETUP.md]: Can we set it up? If not, I will update this section accordingly.

Comment (gudmdharalds) [reply docs/SETUP.md]: Now enabled: https://github.com/Automattic/vip-cli/blob/trunk/.github/dependabot.yml

File: docs/SETUP.md (hunk 3 above)
Comment (gudmdharalds) [reply docs/SETUP.md]: Resolved in 43a15e3.

**Pull request merged**